### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by lucasborin** [SAP/code-pal-for-abap#424]
Currently, the code inspector calls each of the code pal Checks isolated and it depends on singletons (buffers) the avoid multiple database accesses or redundancies. For instance, if we have 50 active checks, the code inspector and code pal will start and run all the 50 checks. 

The `Y_CHECK_BASE` class, which is the base class for all the Checks, performs the setup and determines if the inspection is relevant. For instance, if an object is not relevant because it is out of the creation date range, all the 50 mentioned Checks will perform the same logic and abort the inspection.

Therefore, my proposal is to centralize the code pal in a single Check so that it has only one runtime execution performing the inspection relevance. It means that the code inspector variant will have only one entry, and we could use the Profiles feature to determine which Checks are active. 

Ps: it will force everybody to migrate the customizing from the code inspector variant to the Profile feature. We could also create a tool to perform it. 

In short, for instance: A new Y_CHECK_CODE_PAL class checks all the pre-requisites and instantiate and calls the sub-classes (Checks).

**Comment by eugen-guenther-sap** [comment]
i think its not a good approach, cause it will exclude already existing CI variants.

A rough idea:
Some of the scans running asynchronously, maybe we can store a list of data in memory, limit the list to about 30 objects to avoid memory overflow.

**Comment by lucasborin** [comment]
> i think its not a good approach, cause it will exclude already existing CI variants.
> 
> A rough idea: Some of the scans running asynchronously, maybe we can store a list of data within the memory and limit the list to about 30 objects to avoid a memory overflow.

That's how Code Pal works today, using buffers.

**Comment by D072189** [comment]
Extending the configuration tool to do even more "under the hood" configuration hidden from ATC/SCI is not in our best interest, especially in light of planned moves to the cloud. 

Also, never optimize pre-maturely: Compared against the time necessary to actually check the code of a large program, I would predict the performance losses here are minimal. Unless there are measurement that this is an actual problem in practice, I don't think this is actually a performance problem.
